Works around this error:
locale.c:6237:12: error: use of undeclared identifier '_NL_ADDRESS_POSTAL_FMT'
after this commit:
https://github.com/Perl/perl5/commit/3c45ca2e154f5fdc7ce2a0fdf51b192da3a1ae98

--- a/perl_langinfo.h
+++ b/perl_langinfo.h
@@ -257,7 +257,7 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
  * conflict with some other value, in which case after much gnashing of teeth
  * you will find this comment, and end up having to adjust the numbers.   But
  * glibc values are not (so far) negative */
-#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_ADDRESS)
+#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_ADDRESS) || defined(__ANDROID__)
 #  define _NL_ADDRESS_POSTAL_FMT -58
 #  define _NL_ADDRESS_COUNTRY_NAME -59
 #  define _NL_ADDRESS_COUNTRY_POST -60
@@ -273,7 +273,7 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_IDENTIFICATION)
+#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_IDENTIFICATION) || defined(__ANDROID__)
 #  define _NL_IDENTIFICATION_TITLE -70
 #  define _NL_IDENTIFICATION_SOURCE -71
 #  define _NL_IDENTIFICATION_ADDRESS -72
@@ -292,12 +292,12 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_MEASUREMENT)
+#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_MEASUREMENT) || defined(__ANDROID__)
 #  define _NL_MEASUREMENT_MEASUREMENT -85
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_NAME)
+#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_NAME) || defined(__ANDROID__)
 #  define _NL_NAME_NAME_FMT -86
 #  define _NL_NAME_NAME_GEN -87
 #  define _NL_NAME_NAME_MR -88
@@ -307,13 +307,13 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_PAPER)
+#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_PAPER) || defined(__ANDROID__)
 #  define _NL_PAPER_HEIGHT -92
 #  define _NL_PAPER_WIDTH -93
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
-#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_TELEPHONE)
+#if ! defined(HAS_NL_LANGINFO) || ! defined(LC_TELEPHONE) || defined(__ANDROID__)
 #  define _NL_TELEPHONE_TEL_INT_FMT -94
 #  define _NL_TELEPHONE_TEL_DOM_FMT -95
 #  define _NL_TELEPHONE_INT_SELECT -96
@@ -333,7 +333,8 @@ typedef int nl_item;    /* Substitute 'int' for emulated nl_langinfo() */
  || ! LC_MEASUREMENT_AVAIL_     \
  || ! LC_NAME_AVAIL_            \
  || ! LC_PAPER_AVAIL_           \
- || ! LC_TELEPHONE_AVAIL_
+ || ! LC_TELEPHONE_AVAIL_       \
+ || __ANDROID__
 
 #  define HAS_MISSING_LANGINFO_ITEM_
 #endif
